### DISCUSSION TITLE
` make` error

### DISCUSSION THREAD

**Discussion by MaryamHsn** [cockpit-project/cockpit#16617] [Q&A] [answered]
Cockpit version: 257
OS:   Raspbian GNU/Linux 10 

when I run `make` ,I receive error 

after I clone cockpit from git ,without any change in code, I ran 
`sudo ./autogen.sh --prefix=/usr --enable-debug`.
after `Now type 'make' to compile cockpit. ` message I tried to  `sudo make` and I got below errors. notice without `sudo` Igot permission denied error.

```
pi@raspberrypi:~/cockpit $ sudo make
make  all-am
make[1]: Entering directory '/home/pi/cockpit'
  CCLD     test-packet-channel
/usr/bin/ld: src/bridge/test_packet_channel-test-packet-channel.o: file not recognized: file truncated
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:4003: test-packet-channel] Error 1
make[1]: Leaving directory '/home/pi/cockpit'
make: *** [Makefile:3080: all] Error 2
```

**Comment by jelly** [comment]
It is not required to build the package with `sudo` so my recommendation is to start over from a fresh clone and run

```
./autogen.sh --prefix=/usr
make
```

Also make sure you have enough free space left on the device you are compiling on.

**Comment by martinpitt** [comment, marked as answer]
> file not recognized: file truncated

That sounds like you may either have run out of disk space (check `df -h`) or have some memory or file system corruption (check `dmesg`)?
